### PR TITLE
Fix inconsistent read bug

### PR DESCRIPTION
**Pull request by feeblefakie**
Thank you, Yuji! 🙏 
https://scalar-labs.atlassian.net/browse/DLT-7969

### CODE REVIEW

File: src/integration-test/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitIntegrationTest.java [hunk 1]
```diff
@@ -114,11 +114,30 @@ public void get_CalledTwice_ShouldReturnFromSnapshotInSecondTime()
...
 
     // Act
-    transaction.get(get);
-    transaction.get(get);
+    Optional<Result> result1 = transaction.get(get);
+    Optional<Result> result2 = transaction.get(get);
 
     // Assert
     verify(storage).get(any(Get.class));
+    assertThat(result1).isEqualTo(result2);
```
Comment (feeblefakie) [inline src/integration-test/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitIntegrationTest.java:122]: This is not really related to this PR, but added for consistency with the next test.

**Review by yito88 — DISMISSED** [review]
Thank you! LGTM!
After this fix was applied, the Jepsen test passed.

**Review by brfrn169 — APPROVED** [review]
LGTM! +1

File: src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java [hunk 2]
```diff
@@ -100,6 +100,13 @@ public void put(Snapshot.Key key, Delete delete) {
     deleteSet.put(key, delete);
   }
 
+  public boolean containsKey(Snapshot.Key key) {
+    if (writeSet.containsKey(key) || readSet.containsKey(key)) {
```
[the remark this answers is not in the document]
Comment (feeblefakie) [reply src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java:104]: Thank you! That looks better! I'll fix it.

[the remark this answers is not in the document]
Comment (feeblefakie) [reply src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java]: Updated.

File: src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java [hunk 3]
```diff
@@ -100,6 +100,10 @@ public void put(Snapshot.Key key, Delete delete) {
     deleteSet.put(key, delete);
   }
 
+  public boolean containsKey(Snapshot.Key key) {
+    return writeSet.containsKey(key) || readSet.containsKey(key);
```
Comment (Yonezawa-T2) [inline src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java:104]: Should we check that the key is not in `deleteSet`?

This may be off-topic to this PR, but what happens if we get/scan after delete?

- `get(key1)` → `delete(key1)` → `get(key1)`
- `scan(key1...key10)` → `delete(key5)` → `scan(key1...key10)`

Note that `get` after `write` results in `CrudRuntimeException("reading already written data is not allowed")` raised from `Snapshot#get` while `get` after `delete` doesn't seem to be checked.
